gnomad_qc/v4/sample_qc/interval_qc.py

```diff
@@ -0,0 +1,786 @@
+            'all', 'XX' and 'XY' each is a dict of per platform values.
+
+    :param mt: Input interval coverage MatrixTable.
+    :param platform_ht: Input platform assignment Table.
+    :param mean_dp_thresholds: List of mean DP thresholds to use for computing the fraction of samples with mean
+        interval DP >= the threshold.
+    :param split_by_sex: Whether the interval QC should be stratified by sex. If True, `mt` must be annotated with
+        'sex_karyotype'.
+    :return: Table with interval QC annotations.
+    """
+
+    def _get_agg_expr(expr, agg_func=hl.agg.mean, group_by=None):
+        """
+        Helper function to call `agg_func` on `expr` with optional stratification by sex karyotype and `group_by`.
+
+        :param expr: Expression to pass to `agg_func`.
+        :param agg_func: Function to use for aggregation of `expr`. Default is hl.agg.mean.
+        :return: Aggregation expression.
+        """
+        if group_by is not None:
+            agg_func = hl.agg.group_by(group_by, agg_func(expr))
+        else:
+            agg_func = agg_func(expr)
+        agg_expr = {"all": agg_func}
+        if split_by_sex:
+            agg_expr.update(
+                {
+                    "XX": hl.agg.filter(mt.sex_karyotype == "XX", agg_func),
+                    "XY": hl.agg.filter(mt.sex_karyotype == "XY", agg_func),
+                }
+            )
+
+        return agg_expr
+
+    num_samples = mt.count_cols()
+    mt = mt.annotate_cols(platform=platform_ht[mt.col_key].qc_platform)
+    mt = mt.filter_cols(hl.is_defined(mt.platform))
+    num_samples_no_platform = num_samples - mt.count_cols()
+    if num_samples_no_platform > 0:
+        logger.warning(
+            "Number of samples in MT with no platform assignment: %d",
+            num_samples - mt.count_cols(),
+        )
+
+    # Note: Default hl.vds.interval_coverage will return a list for 'fraction_over_dp_threshold' where the
+    # second element is dp >= 1 (dp > 0)
+    agg_groups = [("", None), ("platform_", mt.platform)]
+    mt = mt.annotate_rows(
+        **{
+            f"{prefix}interval_mean_dp": _get_agg_expr(mt.mean_dp, group_by=group_by)
+            for prefix, group_by in agg_groups
+        },
+        **{
+            f"{prefix}fraction_over_{dp}x": _get_agg_expr(
+                mt.mean_dp >= dp, agg_func=hl.agg.fraction, group_by=group_by
+            )
+            for dp in mean_dp_thresholds
+            for prefix, group_by in agg_groups
+        },
+        **{
+            f"{prefix}mean_fraction_over_dp_0": _get_agg_expr(
+                mt.fraction_over_dp_threshold[1], group_by=group_by
+            )
+            for prefix, group_by in agg_groups
+        },
+    )
+
+    mt = mt.annotate_globals(
+        mean_dp_thresholds=mean_dp_thresholds,
+        platform_n_samples=mt.aggregate_cols(
+            hl.agg.group_by(mt.platform, hl.agg.count())
+        ),
+    )
+
+    return mt.rows()
```

Comment:
yes that is what I want so that the code for determining pass is cleaner. This is in the docstring, but very happy for suggestions on a clearer way to state it. 
```
        - platform_interval_mean_dp - Same as 'interval_mean_dp', but instead of a single value for 'all', 'XX' and
            'XY' each is a dict of per platform values.
```

Reply:
maybe something like " Same as 'interval_mean_dp', but instead of containing a single value, 'all' contains a dictionary of per platform values" and the XX, XY would be optional depending on split_by_sex?